src/Core/Compiler/NodeConverter.php

```diff
@@ -367,9 +375,34 @@ protected function convertExpressionNode(ExpressionNodeInterface $node)
      */
     protected function convertBooleanNode(BooleanNode $node)
     {
-        $stack = $this->convertArrayNode(new ArrayNode($node->getStack()));
-        $initializationPhpCode = '// Rendering Boolean node' . chr(10);
-        $initializationPhpCode .= $stack['initialization'] . chr(10);
+        $booleanStack = $node->getStack();
+
+        // Quick decisions: if there is only one node and it can be determined to be a string/numeric/boolean already
+        // then use it directly.
+        if (count($booleanStack) === 1) {
+            $compiledOnlyNode = $this->convert($booleanStack[0]);
+            // Execution without possible quotation marks, then converted to lowercase for easier comparison.
+            $execution = trim($compiledOnlyNode['execution'], '"\'\\');
+            $lowercaseExecution = strtolower($execution);
+            if (is_numeric($execution) || is_bool($execution)) {
+                $execution = $execution ? 'true' : 'false';
+            } elseif ($lowercaseExecution === 'false' || empty($execution)) {
+                $execution = 'false';
+            } elseif ($lowercaseExecution === 'true') {
+                $execution = 'true';
+            } else {
+                $execution = false;
+            }
+            if ($execution) {
```

Comment:
Can we do an explicit `!empty($execution)` here to make it clear that we don't expect some string magic here to evaluate the `$execution` in place but only check that it's code is not empty?

Reply:
We can do one better and compare `!== false` and add a comment what happens if it *is* `false` :)